### DISCUSSION TITLE
Is there a way to logout without reload using Jestream Livewire?

### DISCUSSION THREAD

**Discussion by macellinoss275** [livewire/livewire#7664] [1. Help] [answered]
So I was able to utilize Livewire 3 navigate to load the pages without reload, but one thing stuck with me. I wasn't able to do so with the Logout button.

Here is how I do the reload using Jestream (Bootstrap version) Livewire:

```
              <li>
                <a class="dropdown-item" href="{{ route('logout') }}" onclick="event.preventDefault(); document.getElementById('logout-form').submit();">
                  <i class='ti ti-logout me-2'></i>
                  <span class="align-middle">Logout</span>
                </a>
              </li>
              <form method="POST" id="logout-form" action="{{ route('logout') }}">
                @csrf
              </form>
```

Even if I give wire:navigate to the anchor like the code below, it's still reload the page when logout:

`<a class="dropdown-item" href="{{ route('logout') }}" onclick="event.preventDefault(); document.getElementById('logout-form').submit();">`

So how do/can I do a logout without any reload just like Vue or React or etc? I was able to do a logout without reload using Inertia Vue in another project, but still couldn't do this with Livewire 3 for this project.

**Comment by danie-ramdhani** [comment, marked as answer]
use button to call logout function instead using `a tag`:

```blade
    <button type="button" wire:click="logout">
        Logout
    </button>
```

```php
    public function logout()
    {
        auth()->logout();

        session()->invalidate();
        session()->regenerateToken();

        $this->redirect(Login::class, navigate: true);
    }
```

**Reply by macellinoss275** [reply]
I can imagine that could work and I should have mentioned this earlier. But a logout button is usually a part of either navbar or sidebar after clicking your profile picture and then it will show a dropdown containing the button.

My logout button is in the navbar and the navbar itself doesn't have any PHP files since it's just a part of the layout that I will call inside my main layout. Because it doesn't have Livewire PHP file, I couldn't do that.

Is there a way around this?

**Reply by kgsint** [reply]
What about extract navbar to it's own livewire component?I think this might work.

**Reply by macellinoss275** [reply]
@kgsint How do I do that? I know how to make a component that's used in the page via url (App\Livewire, resources\views\livewire, and routes/web.php). But I don't know how to do that without any routes. Is it still the same steps?

**Reply by kgsint** [reply]
@macellinoss275 It's just a component and **doesn't have to be a full-page component** so routing is optional in your case I think. And yes, you can follow the same steps except registering route and this should work. I suppose you already know this but just in case, 
- create a livewire component using artisan command `php artisan livewire:make Navbar`
- move navbar and related html elements to `resources/livewire/navbar.blade.php` replace them with`<livewire:navbar />` or `@livewire('navbar')` and just reminder livewire component has to be single root
-  in `app/Livewire/Navbar.php`, you can follow the steps @danie-ramdhani mentioned above
Hope this help!

**Reply by macellinoss275** [reply]
OMG thank you so much guys! @danie-ramdhani @kgsint
